src/tracks/MidiClip.cpp

```diff
@@ -474,6 +474,7 @@ MidiClip * MidiClip::adjacentMidiClipByOffset(int offset) const
 {
 	auto& clips = m_instrumentTrack->getClips();
 	int clipNum = m_instrumentTrack->getClipNum(this);
+	if (clipNum + offset < 0 || clipNum + offset > static_cast<int>(clips.size() - 1)) { return nullptr; }
 	return dynamic_cast<MidiClip*>(clips[clipNum + offset]);
```

Comment:
A couple of suggestions:
* If you add `offset` when initialising `clipNum`, you don't need to repeat `clipNum + offset` three times.
* I think `>=` is a bit simpler and clearer than combining `>` and `- 1`.
* I'd rather cast `clipNum + offset` (now just `clipNum`) than `clips.size()`, since `int` typically has a lower maximum value than `std::size_t` (there's no concern with negative values, since the first part of the condition handles that).
```suggestion
	int clipNum = m_instrumentTrack->getClipNum(this) + offset;
	if (clipNum < 0 || static_cast<std::size_t>(clipNum) >= clips.size()) { return nullptr; }
	return dynamic_cast<MidiClip*>(clips[clipNum]);
```